### PR TITLE
HWKAPM-739 Add JDBC instrumentation rules for OpenTracing java agent.

### PR DESCRIPTION
**Pull request by objectiser**
Including script in dist for using opentracing agent, and fixed issues with using parent span vs span context from byteman rules.

### CODE REVIEW

**Comment by objectiser** [comment]
@pavolloffay Could you review please?

Reviewer notes:
* This PR adds some rules to record JDBC activity
* Also included is a script to make it easier to use.
* Fixed some issues with creating the childOf relationships - byteman rules could not cope with overridden methods, so explicitly named them.

File: client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingManager.java [hunk 1]
```diff
@@ -435,5 +521,14 @@ public long getExpire() {
             return expire;
         }
 
+        /**
+         * This method provides access to variables associated with this
+         * trace.
+         *
+         * @return The variables
+         */
+        public Map<String,Object> getVariables() {
```
Comment (pavolloffay) [inline client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingManager.java]: space after `,`

File: client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingManager.java [hunk 2]
```diff
@@ -384,6 +469,7 @@ public static void reset() {
 
         private Deque<Span> spanStack = new ArrayDeque<>();
         private Deque<String> idStack = new ArrayDeque<>();
+        private Map<String,Object> variables = new HashMap<>();
```
Comment (pavolloffay) [inline client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingManager.java]: space after `,`

File: client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingManager.java [hunk 3]
```diff
@@ -375,6 +406,60 @@ public static void reset() {
...
+        return !path.startsWith("/hawkular/apm") && (path.lastIndexOf('.') <= path.lastIndexOf('/'));
+    }
+
+    /**
+     * This method retrieves a variable associated with the
+     * current trace.
+     *
+     * @return The variable value, or null if not found
+     */
+    public String getVariableAsString(String name) {
```
Comment (pavolloffay) [inline client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingManager.java]: Is there also `getVariable` which returns an `Object`?

Comment (objectiser) [inline client/agent-opentracing/src/main/java/org/hawkular/apm/agent/opentracing/OpenTracingManager.java]: No, as the API only enables tags to be set for string, boolean and number types, and I had problems trying to cast within the byteman rules.

**Comment by objectiser** [comment]
@pavolloffay Thanks!

**Pull request merged**